sys/kern/kern_proc.c

```diff
@@ -2737,6 +2737,14 @@ kern_proc_vmmap_out(struct proc *p, struct sbuf *sb, ssize_t maxlen, int flags)
 			kve->kve_protection |= KVME_PROT_WRITE;
 		if (entry->protection & VM_PROT_EXECUTE)
 			kve->kve_protection |= KVME_PROT_EXEC;
+#ifdef OBJ_NOLOADTAGS
+		if (obj != NULL && (obj->flags & OBJ_NOLOADTAGS) == 0)
+			kve->kve_protection |= KVME_PROT_LOADTAGS;
```

Comment:
Hmm, this argues even further for doing this "right" where these are protections stored in the VM map entry.  That is a bit more work as it means moving the setting of the flags a bit higher when creating the VM map entry, but it also means we could expose direct protections to userland via mmap() and mprotect(), etc.  Though I'm not quite sure what we want to expose to userland in terms of `PROT_*` bits.

Reply:
My proposal is basically:
```
#define PROT_CAP_NONE 0x08
#define PROT_CAP_READ 0x18
#define PROT_CAP_WRITE 0x28
```
`(prot & PROT_CAP_NONE) == 0`implies `PROT_CAP_READ | PROT_CAP_WRITE`. I think that's as close to usable semantics as we're likely to get. You could alternatively go with purely negative semantics.

Reply:
Well, procstat is a bit special in wanting lots of information that most things might not care about (librtld_db is about the only other in-tree consumer of these bits, plus those that transitively use its interfaces), so they don't necessarily need to be part of mmap/mprotect to show up here (but ultimately we do want a way to control them through those interfaces).

Reply:
Also I think it's completely fine to break this interface in future, our thinking will likely evolve over time (though I think something like what you suggest is what we need, but we might want to pick the bits so they're the same as in VM_PROT_*).